client/login/wp-login/login-links.jsx

```diff
@@ -67,26 +69,31 @@ export class LoginLinks extends React.Component {
 	};
 
 	renderBackLink() {
-		const { backTo, translate } = this.props;
-
-		// A backTo prop may be supplied, allowing the back button href to be controlled.
-		if ( backTo ) {
-			const { hostname, protocol } = urlModule.parse( backTo );
+		// If we seem to be in a Jetpack connection flow, provide some special handling
+		// so users can go back to their site rather than WordPress.com
+		const redirectTo = get( this.props, [ 'query', 'redirect_to' ] );
+		if ( redirectTo ) {
+			const { pathname, query: redirectToQuery } = url.parse( redirectTo, true );
+			if ( pathname === '/jetpack/connect/authorize' && redirectToQuery.client_id ) {
+				const returnToSiteUrl = addQueryArgs(
+					{ client_id: redirectToQuery.client_id },
+					'https://jetpack.wordpress.com/jetpack.returntosite/1/'
```

Comment:
We have never redirected from Calypso to the Jetpack server directly - usually Jetpack does it. I wonder if there is a reason for this...

That being said, this looks good to me, I don't have any specific concerns about it.